### ISSUE TITLE
Container build problems due to using d/b for sessions

### ISSUE THREAD

**Issue by toby-griffiths** [api-platform/api-platform#1282]
I'm having a problem bulding my API-Platform containers because I've configured symfony to store sessions in my d/b.

The error reported…
```
+ mkdir -p var/cache var/log
+ composer dump-autoload --classmap-authoritative --no-dev
Generated optimized autoload files (authoritative) containing 6435 classes
+ composer run-script --no-dev post-install-cmd
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!
!!   // Clearing the cache for the prod environment with debug
!!   // false
!!
!!
!!  In AbstractPostgreSQLDriver.php line 73:
!!
!!    An exception occurred in driver: SQLSTATE[08006] [7] could not connect to s
!!    erver: Connection refused
!!      Is the server running on host "localhost" (127.0.0.1) and accepting
!!      TCP/IP connections on port 5432?
!!    could not connect to server: Address not available
!!      Is the server running on host "localhost" (::1) and accepting
!!      TCP/IP connections on port 5432?
!!
!!
!!  In PDOConnection.php line 31:
!!
!!    SQLSTATE[08006] [7] could not connect to server: Connection refused
!!      Is the server running on host "localhost" (127.0.0.1) and accepting
!!      TCP/IP connections on port 5432?
!!    could not connect to server: Address not available
!!      Is the server running on host "localhost" (::1) and accepting
!!      TCP/IP connections on port 5432?
!!
!!
!!  In PDOConnection.php line 27:
!!
!!    SQLSTATE[08006] [7] could not connect to server: Connection refused
!!      Is the server running on host "localhost" (127.0.0.1) and accepting
!!      TCP/IP connections on port 5432?
!!    could not connect to server: Address not available
!!      Is the server running on host "localhost" (::1) and accepting
!!      TCP/IP connections on port 5432?
!!
!!
!!  cache:clear [--no-warmup] [--no-optional-warmers] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command>
!!
!!
Script @auto-scripts was called via post-install-cmd
ERROR: Service 'php' failed to build: The command '/bin/sh -c set -eux;         mkdir -p var/cache var/log;     composer dump-autoload --classmap-authoritative --no-dev;       composer run-script --no-dev post-install-cmd;  chmod +x bin/console; sync' returned a non-zero code: 1
```

Here's the config that's causing the problem…

```yaml
framework:
    session:
        handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler
```

If I comment out the handler_id, then I can bild build the container fine.  It would be great if there was a way to avoid having to do this on every build, if someone clever can advise how we might do that?

**Comment by soyuka** [comment]
what's the actual error? :|

**Comment by toby-griffiths** [comment]
Doh.  Sorry.  I've updated the descrpiption with the build output part showing the errors.

**Comment by toby-griffiths** [comment]
For anyone else coming across this ticket with a similar error, I finally got the bottom of it.

The problem was cause by one of my Twig extensions, several dependencies down the line, loaded the Session service, which in turn, attempted to connect to the d/b as sessions are stored there.

I resolved the issue by simply implementing `Symfony\Contracts\Service\ServiceSubscriberInterface`, so that the dependencies are not loaded when the extension is instantiated.
